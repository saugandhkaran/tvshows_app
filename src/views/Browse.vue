<template>
  <div class="browse">
    <h2 class="flex-container center">Browse from our amazing collection of TV episodes.</h2>
    <tv-show-list>
      <div slot-scope="{ data: TvShowList, loading: loading, error: error }">
        <div v-for="genre in definedGenres" :key="genre.genre" class="flex-container column">
          <h2 class="flex-container">{{genre.label}} TV</h2>
          <section class="flex-container section overflow-x">
            <tv-show-card v-for="show in $options.filters.returnGenreTvShows(TvShowList,genre.genre)" :key="show.id" :showObject="show" @click.native="goToShowDetailsPage(show.id)"></tv-show-card>
          </section>
          <h4 class="flex-container center" v-if="loading">Loading results ...</h4>
          <h4 class="flex-container error center" v-if="error">Something went wrong</h4>
        </div>
      </div>
    </tv-show-list>
  </div>
</template>

<script>
import MovieList from '@/components/renderless/TvShowList.vue'
import TvShowCard from '@/components/TVShowCard.vue'
import definedGenres from './../genres'
export default {
  name: 'Browse',
  components: {
    'tv-show-list': MovieList,
    'tv-show-card': TvShowCard
  },
  data () {
    return {
      definedGenres: definedGenres
    }
  },
  methods: {
    goToShowDetailsPage(id) {
      this.$router.push(`browse/${id}`);
    }
  }
}
</script>

<style lang="scss" scoped>
h2 {
  padding-left: 20px;
}
</style>
