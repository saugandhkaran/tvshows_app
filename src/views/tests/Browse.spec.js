import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import Browse from '../Browse';

const localVue = createLocalVue();
localVue.use(VueRouter)
const router = new VueRouter();

describe('Browse', () => {
  it('should match the Browse html', () => {
    const wrapper = shallowMount(Browse, {
      stubs: {
        'tv-show-list': 'tv-show-list',
      },
    })
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should route to path on click of TV show card', () => {
    const wrapper = shallowMount(Browse, {
      localVue,
      router,
      stubs: {
        'tv-show-list': 'tv-show-list',
      },
      $route: {
        path: '/test'
      }
    });
    wrapper.vm.goToShowDetailsPage('1');
    expect(wrapper.vm.$route.path).toEqual('/browse/1')
  });
});
