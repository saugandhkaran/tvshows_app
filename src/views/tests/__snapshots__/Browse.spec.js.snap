// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Browse should match the Browse html 1`] = `
<div
  class="browse"
>
  <h2
    class="flex-container center"
  >
    Browse from our amazing collection of TV episodes.
  </h2>
   
  <div />
</div>
`;
